app/Http/Controllers/Backend/PermissionsController.php

```diff
@@ -0,0 +1,209 @@
+                       
+            // check save successfull?
+            if ($role->update($arrRole)) {
+                $request->session()->flash('message', trans('messages.edit_role_ok'));
+            } else {
+                $request->session()->flash('message', trans('messages.edit_role_not_ok'));
+            }
+        } else {
+            $request->session()->flash('message', trans('messages.role_exists'));
+        }
+        return redirect()->route('admin.permissions.index');
+    }
+
+    /**
+     * Remove the specified Permissions from storage.
+     *
+     * @param int                      $id      id
+     * @param \Illuminate\Http\Request $request request for delete
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id, Request $request)
+    {
+        $permission = Permission::findOrFail($id);
+
+        // check delete permission ok?
+        if ($permission->delete()) {
+            $request->session()->flash('message', trans('messages.delete_role_ok'));
+        } else {
+            $request->session()->flash('message', trans('messages.delete_role_not_ok'));
+        }
+
+        return redirect()->route('admin.permissions.index');
+    }
+
+    /**
+     * Auto get list Module.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    private function getModule()
+    {
+        $modules = [];
+        $pathController = config('app.path_controllers');
+        $filesInFolder = \File::files($pathController);
+
+        foreach ($filesInFolder as $path) {
+            // get name of controller
+            $controller = pathinfo($path)['filename'];
+            // remove string Controller in controller name
+            $module     = str_replace('Controller', '', $controller);
+            $modules[]  = $module;
+        }
+        // remove module except without array
+        $modules = array_diff($modules, config('app.except_module'));
+        return $modules;
+    }
+
+    /**
+     * Check permission is exists.
+     *
+     * @param \Illuminate\Http\Request $request request
+     *
+     * @return \Illuminate\Http\Response
+     */
+    private function checkPermissionExisted($request)
+    {
+        // set condition where
+        $arrRole = $this->setPermission($request);
+        $check  = Permission::where('module', $request->module)
+            ->where('see', $arrRole['see'])
+            ->where('inset', $arrRole['inset'])
+            ->where('edit', $arrRole['edit'])
+            ->where('destroy', $arrRole['destroy'])
+           ->first();
+        return !is_null($check) ;
+    }
+
+    /**
+     * Set permission.
+     *
+     * @param \Illuminate\Http\Request $request request
+     *
+     * @return \Illuminate\Http\Response
+     */
+    private function setPermission($request)
+    {
+        // set permission
+        $noPermission  = config('app.no_permission');
+        $hasPermission = config('app.has_permission');
+
```

Comment:
define an array of list permission in config 
```php
$permissions  = [
'see', 'insert', 'edit', 'destroy',
]
```
Then loop through  that array and assign value from request
```php
foreach($permissions as $per){
    $arrRole = $request->has($per);
}
return $arrRole
```